### PR TITLE
Backport #1616 to 6.x

### LINKED ISSUE #1617: Back-port #1616 to 6.x
Back-port #1616 to 6.x

### PR DESCRIPTION
**Pull request by Saibamen**
Backport #1616 to 6.x

Fixes #1617

### CODE REVIEW

File: projects/RabbitMQ.Client/client/impl/RecordedBinding.cs [hunk 1]
```diff
@@ -85,7 +85,7 @@ public override string ToString()
 
         public RecordedBinding WithArguments(IDictionary<string, object> value)
         {
-            Arguments = value;
+            Arguments = new Dictionary<string, object>(value);
```
Comment (Saibamen) [inline projects/RabbitMQ.Client/client/impl/RecordedBinding.cs:88]:
```suggestion
            if (value is null)
            {
                Arguments = null;
            }
            else
            {
                Arguments = new Dictionary<string, object>(value);
            }
```

File: projects/RabbitMQ.Client/client/impl/RecordedConsumer.cs [hunk 2]
```diff
@@ -97,7 +97,7 @@ public string Recover(IModel channelToUse)
 
         public RecordedConsumer WithArguments(IDictionary<string, object> value)
         {
-            Arguments = value;
+            Arguments = new Dictionary<string, object>(value);
```
Comment (Saibamen) [inline projects/RabbitMQ.Client/client/impl/RecordedConsumer.cs:100]:
```suggestion
            if (value is null)
            {
                Arguments = null;
            }
            else
            {
                Arguments = new Dictionary<string, object>(value);
            }
```

File: projects/RabbitMQ.Client/client/impl/RecordedExchange.cs [hunk 3]
```diff
@@ -57,7 +57,7 @@ public override string ToString()
 
         public RecordedExchange WithArguments(IDictionary<string, object> value)
         {
-            Arguments = value;
+            Arguments = new Dictionary<string, object>(value);
```
Comment (Saibamen) [inline projects/RabbitMQ.Client/client/impl/RecordedExchange.cs:60]:
```suggestion
            if (value is null)
            {
                Arguments = null;
            }
            else
            {
                Arguments = new Dictionary<string, object>(value);
            }
```

File: projects/RabbitMQ.Client/client/impl/RecordedQueue.cs [hunk 4]
```diff
@@ -66,7 +66,7 @@ protected string NameToUseForRecovery
 
         public RecordedQueue WithArguments(IDictionary<string, object> value)
         {
-            Arguments = value;
+            Arguments = new Dictionary<string, object>(value);
```
Comment (Saibamen) [inline projects/RabbitMQ.Client/client/impl/RecordedQueue.cs:69]:
```suggestion
            if (value is null)
            {
                Arguments = null;
            }
            else
            {
                Arguments = new Dictionary<string, object>(value);
            }
```

File: projects/RabbitMQ.Client/client/impl/RecordedBinding.cs [hunk 5]
```diff
@@ -85,7 +85,14 @@ public override string ToString()
...
-            Arguments = value;
+            if (value is null)
+            {
+                Arguments = null;
+            }
+            else
+            {
+                Arguments = new Dictionary<string, object>(value);
+            }
             return this;
```
Comment (Saibamen) [inline projects/RabbitMQ.Client/client/impl/RecordedBinding.cs:96]:
```suggestion
            }

            return this;
```

File: projects/RabbitMQ.Client/client/impl/RecordedConsumer.cs [hunk 6]
```diff
@@ -97,7 +97,14 @@ public string Recover(IModel channelToUse)
...
-            Arguments = value;
+            if (value is null)
+            {
+                Arguments = null;
+            }
+            else
+            {
+                Arguments = new Dictionary<string, object>(value);
+            }
             return this;
```
Comment (Saibamen) [inline projects/RabbitMQ.Client/client/impl/RecordedConsumer.cs:108]:
```suggestion
            }

            return this;
```

File: projects/RabbitMQ.Client/client/impl/RecordedExchange.cs [hunk 7]
```diff
@@ -57,7 +57,14 @@ public override string ToString()
...
-            Arguments = value;
+            if (value is null)
+            {
+                Arguments = null;
+            }
+            else
+            {
+                Arguments = new Dictionary<string, object>(value);
+            }
             return this;
```
Comment (Saibamen) [inline projects/RabbitMQ.Client/client/impl/RecordedExchange.cs:68]:
```suggestion
            }

            return this;
```

File: projects/RabbitMQ.Client/client/impl/RecordedQueue.cs [hunk 8]
```diff
@@ -66,7 +66,14 @@ protected string NameToUseForRecovery
...
-            Arguments = value;
+            if (value is null)
+            {
+                Arguments = null;
+            }
+            else
+            {
+                Arguments = new Dictionary<string, object>(value);
+            }
             return this;
```
Comment (Saibamen) [inline projects/RabbitMQ.Client/client/impl/RecordedQueue.cs:77]:
```suggestion
            }

            return this;
```

File: projects/Unit/TestConnectionRecovery.cs [hunk 9]
```diff
@@ -757,14 +764,15 @@ public void TestRecoveringConsumerHandlerOnConnection_EventArgumentsArePassedDow
...
-                args.ConsumerArguments["first-argument"] = "event-handler-set-this-value";
+                consumerArgumentMatches = (string)args.ConsumerArguments[key] == value;
             };
 
             CloseAndWaitForRecovery();
             Assert.That(ctagMatches, Is.True, "expected consumer tag to match");
             Assert.That(consumerArgumentMatches, Is.True, "expected consumer arguments to match");
-            Assert.That(myArgs, Does.ContainKey("first-argument").WithValue("event-handler-set-this-value"));
+            Assert.That(arguments.ContainsKey(key), Is.True);
+            string actualVal = arguments[key];
```
Comment (Saibamen) [inline projects/Unit/TestConnectionRecovery.cs:774]:
```suggestion
            string actualVal = (string)arguments[key];
```

**Comment by Saibamen** [comment]
Sorry for mess. GitHub does not allow to commit multiple files via web browser editor or commit multiple PR suggestion in one commit.

**Comment by lukebakken** [comment]
@Saibamen no worries, I cleaned it up.

**Comment by Saibamen** [comment]
`TestQueueRecoveryWithDlxArgument_RabbitMQUsers_hk5pJ4cKF0c` is still missing, but I'm not familiar with NUnit and test folders and helpers was refactored in master branch.

**Review by lukebakken — APPROVED** [review]
Thank you!

**Pull request merged**